test/bdd/pkg/didexchange/didexchange_sdk_steps.go

```diff
@@ -122,20 +122,8 @@ func (d *SDKSteps) validateConnection(agentID, stateValue string) error {
 }
 
 func (d *SDKSteps) approveRequest(agentID string) error {
-	go func() {
-		for e := range d.actionCh[agentID] {
-			switch v := e.Properties.(type) {
-			case didexchange.Event:
-				d.connectionID[agentID] = v.ConnectionID()
-			case error:
-				panic(fmt.Sprintf("Service processing failed: %s", v))
-			}
-
-			clientOpts := d.getClientOptions(agentID)
-			e.Continue(clientOpts)
-		}
-	}()
-
+	// sends the signal which automatically handles events
+	d.nextAction <- struct{}{}
```

Comment:
The approval should be based on the Agent (Alice or Bob). Here it looks like it doesn't matter who is approving. Ideally it should be `d.nextAction[agentID]`.

Reply:
nice catch